opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1644,6 +1698,13 @@ namespace Opm {
                     cellRes[ flowPhaseToEbosCompIdx( flowPhaseIdx ) ] /= refDens;
                     cellRes[ flowPhaseToEbosCompIdx( flowPhaseIdx ) ] *= cellVolume;
                 }
+                if (has_solvent_) {
+                    // no need to store refDens for all cells?
+                    const auto& intQuants = ebosSimulator_.model().cachedIntensiveQuantities(cellIdx, /*timeIdx=*/0);
```

Comment:
Ok. Grep-ing for "cached" seems to indicate that it is only the well model that still has this, it would be best to change it all at once I think, in a separate PR.